editor/src/clj/editor/workspace.clj

```diff
@@ -63,6 +63,13 @@ ordinary paths."
   (^File [workspace path]
    (io/file (project-path workspace) (str (skip-first-char plugins-dir) (skip-first-char path)))))
 
+(defn as-proj-path
+  ^String [workspace file-or-path]
+  (let [file (io/as-file file-or-path)
+        project-directory (project-path workspace)]
+    (when (fs/below-directory? file project-directory)
+      (resource/file->proj-path project-directory file))))
```

Comment:
You might think `file->proj-path` already does what this function does, but it actually returns the absolute path if the file is not below the project. I dare not change it.